spinn_front_end_common/utility_models/data_speed_up_packet_gatherer_machine_vertex.py

```diff
@@ -286,28 +450,556 @@ def get_local_provenance_data(self):
         return prov_items
 
     @staticmethod
-    def set_cores_for_data_extraction(
-            transceiver, extra_monitor_cores_for_router_timeout,
+    def locate_correct_write_data_function_for_chip_location(
+            uses_advanced_monitors, machine, x, y, transceiver,
+            extra_monitor_cores_to_ethernet_connection_map):
+        """ supports other components figuring out which gather and function \
+        to call for writing data onto spinnaker
+
+        :param uses_advanced_monitors: bool saying if the system is using\
+         advanced monitors
+        :param machine: the SpiNNMachine instance
+        :param x: the chip x coordinate to write data to
+        :param y: the chip y coordinate to write data to
+        :param extra_monitor_cores_to_ethernet_connection_map: mapping between\
+         cores and connections
+        :param transceiver: the SpiNNMan instance
+        :return: a write function of either a LPG or the spinnMan
+        :rtype: func
+        """
+        if uses_advanced_monitors:
```

Comment:
The code below this line seems a bit convoluted to me, doing `if …: long-case else: short-case`.

Reply:
eh? the if is tiny, but finding the correct gatherer for the send_data requires a few look ups, whereas SDP is just the SpiNNMan write_memory. 

not sure what your trying to hint at here